packages/infra/lib/surescripts-stack.ts

```diff
@@ -0,0 +1,527 @@
+    });
+    this.synchronizeSftpLambda = synchronizeSftp.lambda;
+
+    const sendPatientRequest = this.setupSendPatientRequest({
+      ...commonConfig,
+      surescriptsReplicaBucket: surescriptsReplicaBucket,
+    });
+    this.sendPatientRequestLambda = sendPatientRequest.lambda;
+    this.sendPatientRequestQueue = sendPatientRequest.queue;
+
+    const receiveVerificationResponse = this.setupReceiveVerificationResponse({
+      ...commonConfig,
+      surescriptsReplicaBucket: surescriptsReplicaBucket,
+    });
+    this.receiveVerificationResponseLambda = receiveVerificationResponse.lambda;
+    this.receiveVerificationResponseQueue = receiveVerificationResponse.queue;
+
+    const receiveFlatFileResponse = this.setupReceiveFlatFileResponse({
+      ...commonConfig,
+      surescriptsReplicaBucket: surescriptsReplicaBucket,
+      surescriptsBundleBucket: this.surescriptsBundleBucket,
+    });
+    this.receiveFlatFileResponseLambda = receiveFlatFileResponse.lambda;
+    this.receiveFlatFileResponseQueue = receiveFlatFileResponse.queue;
+  }
+
+  private setupConnectSftp(ownProps: {
+    lambdaLayers: LambdaLayers;
+    vpc: ec2.IVpc;
+    envType: EnvType;
+    sentryDsn: string | undefined;
+    alarmAction: SnsAction | undefined;
+    surescripts: EnvConfig["surescripts"];
+  }): { lambda: Lambda } {
+    const { lambdaLayers, vpc, envType, sentryDsn, alarmAction, surescripts } = ownProps;
+    const { name, entry, lambda: lambdaSettings } = settings.connectSftp;
+
+    const lambda = createLambda({
+      ...lambdaSettings,
+      stack: this,
+      name,
+      entry,
+      envType,
+      envVars: {
+        ...surescriptsEnvironmentVariables({
+          surescripts,
+          allowReplicaAccess: false,
+          allowBundleAccess: false,
+        }),
+        ...(sentryDsn ? { SENTRY_DSN: sentryDsn } : {}),
+      },
+      layers: [lambdaLayers.shared],
+      vpc,
+      alarmSnsAction: alarmAction,
+    });
+
+    return { lambda };
+  }
+
+  private setupSynchronizeSftp(ownProps: {
+    lambdaLayers: LambdaLayers;
+    vpc: ec2.IVpc;
+    envType: EnvType;
+    sentryDsn: string | undefined;
+    alarmAction: SnsAction | undefined;
+    surescriptsReplicaBucket: s3.Bucket;
+    surescripts: EnvConfig["surescripts"];
+  }): { lambda: Lambda; queue: Queue } {
+    const {
+      lambdaLayers,
+      vpc,
+      envType,
+      sentryDsn,
+      alarmAction,
+      surescriptsReplicaBucket,
+      surescripts,
+    } = ownProps;
+    const {
+      name,
+      entry,
+      lambda: lambdaSettings,
+      queue: queueSettings,
+      eventSource: eventSourceSettings,
+    } = settings.synchronizeSftp;
+
+    const lambda = createLambda({
+      ...lambdaSettings,
+      stack: this,
+      name,
+      entry,
+      envType,
+      envVars: {
+        ...surescriptsEnvironmentVariables({
+          surescripts,
+          allowReplicaAccess: true,
+          allowBundleAccess: false,
```

Comment:
I think this is error prone: we define access to the buncket here (env vars) and totally unrelated on like 329 (actual access)...

```ts
surescriptsReplicaBucket.grantReadWrite(lambda);
```

I'd rather have these be build on the basis whether the bucket is set or not on the parameters. Then we have it set only once per function.